src/chrome/chromeDebugSession.ts

```diff
@@ -131,6 +140,8 @@ export class ChromeDebugSession extends LoggingDebugSession {
                     reportFailure(e);
                 }
                 this.failedRequest(request.command, response, e);
+            } finally {
+                this.Events.emitStepStarted(`WaitingAfter.ClientRequest.${request.command}`);
```

Comment:
Would it make more sense to emit that a step is complete, and have the emitter create the WaitingAfter events?

Reply:
I think that conceptually it does make more sense to emit when a step is completed.

In practice, some steps end sometimes in different locations due to multiple return statements or other strategies, and most steps usually begin in a single place, so it seemed simpler to track step beginnings rather than steps ends...

Ideally we should track both being and end with something like recordStep(() = doStepHere());. But that requires a much bigger code change, so I think that the simpler approach is better at this time. What do you think?

Reply:
Or, the emitter can fire some event that says the step is complete, and the reporter could write that out as a WaitingAfter step if it wants to. Would that be more natural? If not you can ignore this feedback.

Reply:
I don't think it should know about "ClientRequests". I see that some steps just end when the next one starts, and these have an explicit end.

Is it a problem that steps end with the next one? What about overlapping requests from the client?

Also, why do we need to track the timings here, doesn't the reportTelemetry method with the callback do that?

Reply:
I'm sorry, I should've sent the two PRs at the same time to give you the full context.
This PR works in conjunction with: https://github.com/Microsoft/vscode-chrome-debug/pull/610
That is where we turn it off, etc...

Reply:
I still don't see how it's turned off in chrome-debug. Also I don't think it makes sense to start logging from -core and only turn it off from the subclass. Then it will log forever for anything besides chrome-debug that doesn't reach up and turn it off.

If this is all for chrome-debug, does any of it make sense for node-debug or any other consumers? If not, we need to rethink this approach.

I also still don't understand why the telemetry that times each request isn't sufficient for the part of this that measures requests. It's the right model for this sort of thing. setBreakpoints, and setExceptionBreakpoints won't come necessarily come one at a time as steps. They could overlap, and then I think your timings would be confusing.